Add new params to configuration: parathreads and target_fullness

tmpolaczyk opened the pull request:
Add these new params to pallet_configuration, and extrinsics to change their value:

```
                        collators_per_parathread: 1,
                        parathreads_per_collator: 1,
                        target_container_chain_fullness: Perbill::from_percent(80),
```

Their functionality is not implemented yet, that will come in future PRs

tmpolaczyk commented on test/moonwall.config.json:
```diff
@@ -9,7 +9,7 @@
             "envVars": ["DEBUG_COLORS=1"],
             "testFileDir": ["suites/common-all", "suites/common-xcm", "suites/common-tanssi", "suites/dev-tanssi"],
-            "multiThreads": true,
+            "multiThreads": false,
```
Undo

girazoki commented on pallets/configuration/src/lib.rs:
```diff
@@ -71,6 +71,14 @@ pub struct HostConfiguration {
+    // the same time, how many parathreads.
+    pub parathreads_per_collator: u32,
+    // % of collators that we expect to be assigned to container chains. Affects fees.
+    // TODO: I guess container chains and parathreads will be counted together
```
We will need to see this, perhaps the count will be parachain based and we assign a percentage to each parathread (like each parathread counts 0.25 of a parachain)

girazoki commented on pallets/configuration/src/lib.rs:
```diff
@@ -71,6 +71,14 @@ pub struct HostConfiguration {
     pub full_rotation_period: u32,
+    // Like collators_per_container but for parathreads
+    pub collators_per_parathread: u32,
+    // TODO: use a less confusing name? this is for when we support assigning 1 collator to more than 1 parathread at
```
`parathread_assignments_per_collator`? but its longer :(

tmpolaczyk replied:
It would be fun to call this `collator_multithreading` but that's also confusing. Anyway renaming fields shouldn't require a migration, so I am leaving this as `parathreads_per_collator` for now.

girazoki commented on runtime/flashbox/src/migrations.rs:
```diff
@@ -20,16 +20,137 @@
+    sp_core::RuntimeDebug,
+    scale_info::TypeInfo,
+)]
+struct HostConfigurationV1 {
```
At some point we should create a runtime/common folder..but for a future PR

girazoki commented on pallets/configuration/src/lib.rs:
```diff
@@ -113,6 +128,9 @@ impl HostConfiguration {
         if self.max_orchestrator_collators < self.min_orchestrator_collators {
             return Err(InconsistentError::MaxCollatorsLowerThanMinCollators);
         }
+        if self.collators_per_parathread != 1 || self.parathreads_per_collator != 1 {
```
love this

girazoki commented on runtime/flashbox/src/migrations.rs:
```diff
@@ -123,10 +244,12 @@ where
         let migrate_services_payment =
             MigrateServicesPaymentAddCredits::<Runtime>(Default::default());
         let migrate_boot_nodes = MigrateBootNodes::<Runtime>(Default::default());
+        let migrate_config2 = MigrateConfigurationParathreads::<Runtime>(Default::default());
```
```suggestion
        let migrate_config_parathread_params = MigrateConfigurationParathreads::<Runtime>(Default::default());
```

girazoki reviewed (approved):
Provided that we add some ts-tests, looks good to me!

tmpolaczyk commented:
Added a typescript test for `target_container_chain_fullness`, the other params are not tested because I added a validation so they cannot be changed.
